gnucash/report/report-core.scm

```diff
@@ -375,14 +375,7 @@ not found.")))
     (let ((options (if (null? rest)
                        (gnc:report-template-new-options template)
                        (car rest))))
-      (gnc:report-set-options! r options)
-      (gnc:options-register-callback
-       #f #f
-       (lambda ()
-         (gnc:report-set-dirty?! r #t)
-         (let ((cb (gnc:report-template-options-changed-cb template)))
-           (if cb (cb r))))
-       options))
+      (gnc:report-set-options! r options))
```

Comment:
:+1: I have no idea what the callback functions are meant to do!

Reply:
That one was to force a redraw of the options dialog if the report template somehow managed to be changed while the dialog was open. How that might have happened is an interesting question but I don't think it's possible in today's GnuCash. The other one was to sync the UI with programmatic options changes. I don't think there are any, but I also think that there are signals in dialog-options.cpp that will handle it.